### PR TITLE
chore(dependencies): increment min stencil version

### PR DESCRIPTION
**Pull request by rwaskiewicz**
## Pull request checklist

Please check if your PR fulfills the following requirements:
- [ ] Docs have been reviewed and added / updated if needed (for bug fixes / features)
- [x] Build (`npm run build`) was run locally and any changes were pushed
- [x] Tests (`npm test`) were run locally and passed
- [x] Prettier (`npm run prettier`) was run locally and passed

## Pull request type

 

 

Please check the type of change your PR introduces:
- [ ] Bugfix
- [x] Feature
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] Documentation content changes
- [ ] Other (please describe):

## What is the current behavior?

GitHub Issue Number: N/A

## What is the new behavior?

this commit increments the minor version of @stencil/core in the peer
dependencies of the package. this change is a result of the stencil team
no longer officially supporting stencil v1.x

BREAKING CHANGE: Drops support for v1.X of Stencil. Stencil projects using
v1.x of `@stencil/core` will need to [migrate to v2.x](https://github.com/ionic-team/stencil/blob/main/BREAKING_CHANGES.md#breaking-changes)

## Does this introduce a breaking change?

- [X] Yes
- [ ] No

Applications using this library will need to migrate to Stencil v2.  

## Testing

- Unit tests were run/passed

### CODE REVIEW

**Review by tanner-reits — APPROVED** [review]
Change looks good, just curious how you're gonna handle the breaking change. Should this go to a v2 branch? This is one of the packages we're going to continue maintaining, right?

**Comment by rwaskiewicz** [comment]
@tanner-reits I'm going to cut a v2.0.0 release after this lands, rather than creating a v2 branch. That way, folks can upgrade with minimal friction (ideally), and we can signal that moving forward, the maintenance we do on this package will only be in conjunction with Stencil v2

File: package.json [hunk 1]
```diff
@@ -22,20 +22,20 @@
...
     "@ionic/prettier-config": "^2.0.0",
     "@rollup/plugin-node-resolve": "^13.0.5",
-    "@stencil/core": "^2.8.1",
+    "@stencil/core": "^2.17.3",
     "@types/jest": "^26.0.15",
     "@types/node": "^14.17.18",
     "@types/sass": "^1.16.0",
     "jest": "^26.6.3",
     "np": "^7.0.0",
+    "prettier": "^2.2.1",
```
Comment (alicewriteswrongs) [inline package.json:36]: 🔡 💯 👍

**Review by alicewriteswrongs — APPROVED** [review]
lgtm 👍! sorry I missed the notification about this one for a bit

**Pull request merged**